ariadne/core/src/Ariadne/UX/PasswordManager.hs

```diff
@@ -78,10 +85,10 @@ createPasswordManager = buildComponent_ "Password Manager" $ do
         voidPassword walletId = atomicModify_ passMapVar $ Map.delete walletId
 
         getPasswordWithUI :: RequestPasswordToUi -> GetPassword
-        getPasswordWithUI requestToUI requestMode walletId = do
+        getPasswordWithUI requestToUI requestMode getExisting walletId = do
             passMap <- readIORef passMapVar
-            case Map.lookup walletId passMap of
-                Just (password, _) -> do
+            case (Map.lookup walletId passMap, getExisting) of
+                (Just (password, _), GetFromMap) -> do
```

Comment:
Super-minor, but `lookup` can be avoided in case of `GetPasswordFromUI` (and you enforce it here with pattern matching on its result).